tests/rbd_mirror/test_cg_mirror.py

```diff
@@ -0,0 +1,397 @@
+Step 19: Repeat above on EC pool
+
+"""
+
+import json
+from copy import deepcopy
+
+from ceph.rbd.initial_config import initial_mirror_config
+from ceph.rbd.utils import check_data_integrity, getdict, random_string
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.group_mirror import (
+    enable_group_mirroring_and_verify_state,
+    group_mirror_status_verify,
+    wait_for_idle,
+)
+from ceph.rbd.workflows.krbd_io_handler import krbd_io_handler
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_group_mirroring(
+    rbd_primary,
+    rbd_secondary,
+    client_primary,
+    client_secondary,
+    primary_cluster,
+    secondary_cluster,
+    pool_types,
+    **kw
+):
+    """
+    Test user can successfully mirror rbd group containing rbd images between primary and secondary sites
+    Args:
+        rbd_obj: RBD object
+        client : client node object
+        **kw: any other arguments
+    """
+
+    for pool_type in pool_types:
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = deepcopy(getdict(rbd_config))
+
+        for pool, pool_config in multi_pool_config.items():
+            group_ls_kw = {}
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+            group_ls_kw.update({"pool": pool})
+
+            for image, image_config in pool_config.items():
+                if "image" in image:
+                    # Add data to the images
+                    err = run_IO(rbd_primary, client_primary, pool, image, **kw)
+                    if err:
+                        return 1
+
+            # Get Groups in a Pool
+            (group, _) = rbd_primary.group.list(**group_ls_kw)
+            log.info("Groups present in pool " + pool + " are: " + group)
+            if _:
+                log.error("Listing Group in a pool Failed")
+                return 1
+            group_ls_kw.update({"group": group.strip()})
+
+            # Get Group Mirroring Status
+            (group_mirror_status, _) = rbd_primary.mirror.group.status(**group_ls_kw)
+            if _:
+                if "mirroring not enabled on the group" in _:
+                    mirror_state = "Disabled"
+                else:
+                    log.error("Getting group mirror status failed : " + str(_))
+                    return 1
+            else:
+                mirror_state = "Enabled"
+            log.info("Group " + group + " mirroring state is " + mirror_state)
+
+            # Enable Group Mirroring and Verify
+            if mirror_state == "Disabled":
+                out = enable_group_mirroring_and_verify_state(
+                    rbd_primary, **group_ls_kw
+                )
+                if out:
+                    log.error("Enabling group mirroring failed")
+                    return 1
+
+            # Wait for group mirroring to complete
+            out = wait_for_idle(rbd_primary, **group_ls_kw)
```

Comment:
can we check mirror complete with status check similar to this 
https://github.com/red-hat-storage/cephci/blob/main/ceph/rbd/workflows/rbd_mirror.py#L44